app/screens/channel/channel_post_list/channel_post_list.tsx

```diff
@@ -55,14 +55,14 @@ const ChannelPostList = ({
     }, [isCRTEnabled, posts, channelId, serverUrl, appState === 'active']);
 
     const onEndReached = useCallback(debounce(async () => {
-        if (!fetchingPosts.current && canLoadPosts.current && posts.length) {
+        if (!fetchingPosts.current && canLoadPosts.current) {
             fetchingPosts.current = true;
             const lastPost = posts[posts.length - 1];
-            const result = await fetchPostsBefore(serverUrl, channelId, lastPost.id);
+            const result = await fetchPostsBefore(serverUrl, channelId, lastPost?.id || '');
             fetchingPosts.current = false;
             canLoadPosts.current = false;
             if (!('error' in result)) {
-                canLoadPosts.current = (result.posts?.length ?? 1) > 0;
+                canLoadPosts.current = (result.posts?.length ?? 0) > 0;
```

Comment:
this is fine, worst case scenario, when opening a channel that has not enough posts loaded, this will trigger twice